### PR TITLE
Signup, Login, Signout integration

### LINKED ISSUE #12: Provide a method of authenticating future requests
- Tokens?
- Store user details and pass through with each request?

Need to decide on approach

### LINKED ISSUE #13: Authorise user login
- Allow authorised users to enter
- Prevent unauthorised logins

Depends on #10 & #24

### LINKED ISSUE #14: Create new user account
After user submits signup form
- Send data to firebase
- Get confirmation
- Redirect to login

Depends on #24 & #11

### LINKED ISSUE #99: Sign out from nav bar
Depends on #24

### PR DESCRIPTION
**Pull request by max-gurr**
- Linked signup component to firebase signup
- Linked login component to firebase signin
- Linked navbar logout icon to firebase signout
- Altered privateroute component
 - Checks if user is signed in
 - If they are, create a Route to the supplied routePath, with all children rendered inside it
 - If they aren't, redirect to login page

At the moment I'm using alerts to notify the user if the signup/login went bad. This can be a point of improvement

### CODE REVIEW

**Review by YungiKim1999 — APPROVED** [review]
Nice modification for the base PrivateRoute component.
LGTM

File: src/pages/loginView.js [hunk 1]
```diff
@@ -33,9 +35,12 @@ export default function LoginView() {
     if (validLogin) {
       // TODO: Call firebase login method with email and password
```
Comment (dongmeilim) [inline src/pages/loginView.js]: no longer needed?

**Review by dongmeilim — APPROVED** [review]
firebase-related stuff looks good to me!

File: src/pages/loginView.js (hunk 1 above)
Comment (max-gurr) [reply src/pages/loginView.js]: good spot!

**Pull request merged**